lib/Element.js

```diff
@@ -233,6 +237,14 @@ Element.prototype.getElementById = function (id) {
   }
 }
 
+Element.prototype.getBoundingClientRect = function () {
+  if (!this.component) {
+    return undefined
+  }
+
+  return this.component.getBoundingClientRect()
+}
```

Comment:
Ah, I see, we could always fake that though. Just test the logic. A full
DOM test shouldn't be required since that varies browser to browser anyway.

On 21 June 2016 at 18:51, Alex Yuly <notifications@github.com> wrote:

> In lib/Element.js
> <https://github.com/Olical/react-faux-dom/pull/52#discussion_r67917889>:
>
> > @@ -233,6 +237,14 @@ Element.prototype.getElementById = function (id) {
> >    }
> >  }
> >
> > +Element.prototype.getBoundingClientRect = function () {
> > +  if (!this.component) {
> > +    return undefined
> > +  }
> > +
> > +  return this.component.getBoundingClientRect()
> > +}
>
> Thanks for your reply. "The bounding rect is something computed after the
> browser has rendered the element" <- Yes, that is correct. This is working
> for me (i.e. it returns the real DOM element's ClientRect) just as soon as
> React has mounted the element to the DOM. In Element's constructor, I added
> a ref property, which gets added to the React element as its ref
> attribute... after mounting, React passes the DOM element to the callback
> function which sets it as a property on the Element instance. Anyway to
> unit test, I think I'll need to fake the DOM... Any suggestions?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Olical/react-faux-dom/pull/52/files/c72c8621c808065401c650170b65fd7c5d9859cc#r67917889>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AATPXcV58c02zF6b96A2jrDOhga7cIvhks5qOCSXgaJpZM4I6_Zy>
> .
>